tracer/src/Datadog.Trace/Iast/Aspects/System.Web.Extensions/JavaScriptSerializerAspects.cs

```diff
@@ -0,0 +1,94 @@
+// <copyright file="JavaScriptSerializerAspects.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+
+#nullable enable
+
+#if NETFRAMEWORK
+
+using System;
+using System.Collections.Generic;
+using Datadog.Trace.DuckTyping;
+using Datadog.Trace.Iast.Dataflow;
+using Datadog.Trace.Logging;
+
+namespace Datadog.Trace.Iast.Aspects.System.Web.Extensions;
+
+/// <summary> JavaScriptSerializer class aspect </summary>
+[AspectClass("System.Web.Extensions")]
+[global::System.ComponentModel.Browsable(false)]
+[global::System.ComponentModel.EditorBrowsable(global::System.ComponentModel.EditorBrowsableState.Never)]
+public class JavaScriptSerializerAspects
+{
+    private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<JavaScriptSerializerAspects>();
+
+    /// <summary>
+    /// DeserializeObject aspect
+    /// </summary>
+    /// <param name="instance"> The instance </param>
+    /// <param name="input"> the json string </param>
+    /// <returns> The target url </returns>
+    [AspectMethodReplace("System.Web.Script.Serialization.JavaScriptSerializer::DeserializeObject(System.String)")]
+    public static object? DeserializeObject(object instance, string input)
+    {
+        IJavaScriptSerializer? serializer;
+        try
+        {
+            serializer = instance.DuckCast<IJavaScriptSerializer>();
+        }
+        catch (Exception ex)
+        {
+            Log.Warning(ex, "Error while casting JavaScriptSerializer");
```

Comment:
This should log an error, because it's very likely that if this catch is hit, the app behavior will be changed.